src/ipc/connectionStatus.ts

```diff
@@ -0,0 +1,22 @@
+import { sleep } from './_utils';
+
+export enum ConnectionStatus {
+  CONNECTED_TO_PUBLIC_GATEWAY,
+  CONNECTED_TO_COURIER,
+  DISCONNECTED_FROM_PUBLIC_GATEWAY,
+  DISCONNECTED_FROM_ALL,
+}
+
+export async function* pollConnectionStatus(): AsyncIterable<ConnectionStatus> {
```

Comment:
I believe reactive programming is more common in UI apps, but I have zero experience with it, so I'm using this pattern. Please feel free to replace with a reactive approach.

Reply:
No worries. Traditional polling (e.g., while/sleep) is a pain for UI threads but await/async stuff should be easy enough to integrate.

Reply:
Great!